src/components/TextField/index.tsx

```diff
@@ -20,10 +21,16 @@ export interface TextFieldProps
   iconLeft?: React.ReactNode;
   iconRight?: React.ReactNode;
   styleType?: Style;
+  className?: string;
 }
 
-const generateClassName = ({ errors, iconLeft, styleType }: IClassNameArgs) => {
-  const baseClass = "input__field";
+const generateClassName = ({
+  errors,
+  iconLeft,
+  styleType,
+  className,
+}: IClassNameArgs) => {
+  const baseClass = `input__field ${className}`;
```

Comment:
For an empty `className` we want to trim the whitespaces
```suggestion
  const baseClass = `input__field ${className}`.trim();
```